spec/features/staff/users_can_view_an_activity_as_xml_spec.rb

```diff
@@ -10,88 +10,14 @@
-          expect(xml.at("iati-activity/budget/@type").text).to eq("1")
-          expect(xml.at("iati-activity/budget/@status").text).to eq("1")
-          expect(xml.at("iati-activity/budget/value").text).to eq(budget.value.to_s)
-          expect(xml.at("iati-activity/budget/period-start/@iso-date").text).to eq(budget.period_start_date.strftime("%Y-%m-%d"))
-          expect(xml.at("iati-activity/budget/period-end/@iso-date").text).to eq(budget.period_end_date.strftime("%Y-%m-%d"))
-        end
+        it_behaves_like "valid activity XML"
       end
 
       context "when the activity is a programme activity" do
         let(:activity) { create(:programme_activity, organisation: organisation, identifier: "IND-ENT-IFIER") }
         let(:xml) { Nokogiri::XML::Document.parse(page.body) }
 
-        it "contains the activity XML" do
-          visit organisation_activity_path(organisation, activity, format: :xml)
-          expect(xml.at("iati-activity/@default-currency").text).to eq(activity.default_currency)
-          expect(xml.at("iati-activity/iati-identifier").text).to eq(activity.identifier)
-        end
-
-        it "contains the funding organisation XML" do
-          visit organisation_activity_path(organisation, activity, format: :xml)
-          expect(xml.at("iati-activity/participating-org[@role = '1']/@ref").text).to eq(activity.funding_organisation_reference)
-          expect(xml.at("iati-activity/participating-org[@role = '1']/@type").text).to eq(activity.funding_organisation_type)
-          expect(xml.at("iati-activity/participating-org[@role = '1']/narrative").text).to eq(activity.funding_organisation_name)
-        end
-
-        it "contains the accountable organisation XML" do
-          visit organisation_activity_path(organisation, activity, format: :xml)
-          expect(xml.at("iati-activity/participating-org[@role = '2']/@ref").text).to eq(activity.accountable_organisation_reference)
-          expect(xml.at("iati-activity/participating-org[@role = '2']/@type").text).to eq(activity.accountable_organisation_type)
-          expect(xml.at("iati-activity/participating-org[@role = '2']/narrative").text).to eq(activity.accountable_organisation_name)
-        end
-
-        it "contains the budget XML" do
-          budget = create(:budget, activity: activity)
-          visit organisation_activity_path(organisation, activity, format: :xml)
-          expect(xml.at("iati-activity/budget/@type").text).to eq("1")
-          expect(xml.at("iati-activity/budget/@status").text).to eq("1")
-          expect(xml.at("iati-activity/budget/value").text).to eq(budget.value.to_s)
-          expect(xml.at("iati-activity/budget/period-start/@iso-date").text).to eq(budget.period_start_date.strftime("%Y-%m-%d"))
-          expect(xml.at("iati-activity/budget/period-end/@iso-date").text).to eq(budget.period_end_date.strftime("%Y-%m-%d"))
-        end
+        it_behaves_like "valid activity XML"
```

Comment:
I've just read about this - pretty clever! 

I wonder if the file that contains it could be called `behaves_like_valid_activity_xml_spec.rb', just so I can understand the link if I have not seen the pattern before - or is that not the convention for this?

Reply:
Agree